### PR TITLE
Post2 release for addressing bugs with the latest release

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: 0.10.0.post2

## Release details

Autonomy:
- Updates the `autonomy publish` command to avoid publishing temporary files
- Adds support for specifying owner when minting a component using `--owner` flag on `autonomy mint` command
- Makes error messages on the on chain dependency checks more user friendly
- Updates the runtime tendermint override update logic to account for `--n` flag
- Updates the runtime setup override update logic to set parameters as defined data types instead of a `list` of the said data type

Packages:
- Adds support for defining owner independent of the transaction sender when minting component on service manager and component registries manager contracts

Fixes:

- https://github.com/valory-xyz/open-autonomy/issues/1901
- https://github.com/valory-xyz/open-autonomy/issues/1902

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] I've updated the dependencies versions to the latest, wherever is possible.
- [ ] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [ ] I built the documentation and updated it with the latest changes
- [ ] I've added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

**Review by angrybayblade — COMMENTED** [review]
@DavidMinarsch Use `--owner` flag when minting the components to define the owner

File: autonomy/chain/config.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ class ChainConfigs:  # pylint: disable=too-few-public-methods
         chain_id=DEFAULT_LOCAL_CHAIN_ID,
     )
 
-    custom = ChainConfig(
+    custom_chain = ChainConfig(
```
Comment (angrybayblade) [inline autonomy/chain/config.py:78]: Addresses https://github.com/valory-xyz/open-autonomy/issues/1901 @Adamantios

File: autonomy/chain/utils.py [hunk 2]
```diff
@@ -95,10 +117,10 @@ def verify_component_dependencies(
             ledger_api=ledger_api,
             token_id=dependency_id,
         )
-        component_public_id = PublicId.from_str(component_metadata["name"]).to_any()
+        component_public_id = _parse_public_id_from_metadata(component_metadata["name"])
```
Comment (angrybayblade) [inline autonomy/chain/utils.py:120]: Addresses https://github.com/valory-xyz/open-autonomy/issues/1902 @Adamantios

File: autonomy/deploy/base.py [hunk 3]
```diff
@@ -319,7 +319,7 @@ def _update_tendermint_params(
                 param_args[TENDERMINT_COM_URL_PARAM] = TENDERMINT_COM.format(idx)
 
         try:
-            if self.service.number_of_agents == 1:
+            if self.service.number_of_agents == 1 and not has_multiple_overrides:
```
Comment (angrybayblade) [inline autonomy/deploy/base.py:322]: One more issue I found while playing around with `from-token` command

**Review by DavidMinarsch — DISMISSED** [review]
Please update the PR description with the notes and also with the issue ids it fixes

**Comment by dagacha** [comment]
Are we introducting any breaking changes in this post release @angrybayblade ?

**Comment by angrybayblade** [comment]
> Are we introducting any breaking changes in this post release @angrybayblade ?

No

File: HISTORY.md [hunk 4]
```diff
@@ -1,5 +1,17 @@
 # Release History - `open-autonomy`
 
+# 0.10.0.post2 (2023-03-30)
+
+Autonomy:
+- Updates the `autonomy publish` command to not avoid publishing temporary files
```
Comment (Adamantios) [inline HISTORY.md:6]:
```suggestion
- Updates the `autonomy publish` command to avoid publishing temporary files
```

**Review by 0xArdi — APPROVED** [review]

**Review by Adamantios — APPROVED** [review]

**Pull request merged**